### PR TITLE
Fix UFS replay when using active LSM

### PR DESCRIPTION
**Pull request by dustinswales**
This PR includes the following changes to enable Replay:
- Always do cold-start for UFS replay with active LSM. If using model initial conditions and forcing from the UFS (e.g. Replay mode) with an active LSM, the SCM performs a cold-start initialization for the LSM. That is, UFS replay mode currently DOES NOT support cases generated from UFS Restart files (e.g spun up (warm) LSM parameters). [Associated physics PR to fix bug](https://github.com/NCAR/ccpp-physics/pull/1069).

- Fix for cases when grid orientation cannot be determined. For testing, I generated SCM cases with a regular 2deg grid that covered CONUS. Some of the locations for these cases fell at locations where the regridding software couldn't determine the orientation. Recall we have IC and UFS history files on different grids, and there are locations where this remapping fails if you pick a point on the edge of a tile.

- Add functionality to replay script to use input file containing longitude/latitude of points to generate SCM cases from. There is a simple script to help create this file.

- Bring in grid area from DEPHY metadata, instead of SCM case configuration file. (I guess in theory all case configuration info should come in via DEPHY file metadata, and the SCM case configuration file would go away? If we go to all DEPHY at some point this would clean up the repository and SCM workflow)

@grantfirl 
@hertneky @mzhangw

### CODE REVIEW

**Comment by dustinswales** [comment]
@grantfirl This is the branch I've been working with for my Replay debugging if you wanted to take a look.

**Comment by grantfirl** [comment]
@dustinswales Has this been tested using the new module-based setup on Hera? I'm trying to do this (setting up the env_ufsreplay conda environment) and not having luck on Hera. Conda gets stuck on "solving environment".

**Comment by hertneky** [comment]
@grantfirl @dustinswales I will chime in that I had this issue before the module-based setup. I ended up having to do it manually on Hera. Worked okay on Cheyenne/Derecho I think.

**Comment by grantfirl** [comment]
@dustinswales I'd like to be able to test and merge this. Can you please update your PR branch to the latest main?

**Comment by grantfirl** [comment]
@dustinswales For posterity and ease of review, could you expand the description? (I.e. what was the problem and how it was fixed)

**Comment by dustinswales** [comment]
@grantfirl Done. Ready for review.

**Comment by grantfirl** [comment]
@dustinswales Thanks, Dustin. Looking at the ccpp/physics changes, it looks like the changes are based on if https://github.com/NCAR/ccpp-scm/pull/439 is already merged? I'm just confused seeing the ccpp/physics changes include more than just the GFS_phys_time_vary.scm.F90 file.

**Comment by grantfirl** [comment]
@dustinswales OK, nevermind, I think. IIRC, there were some merges into ccpp/physics that didn't have associated SCM PRs (since nothing changed in *_typedefs), which is why it is showing more changes in ccpp/physics than it otherwise would. I think that is what is going on.

I'm thinking about merging https://github.com/NCAR/ccpp-scm/pull/439 first, though, just to make sure...

**Comment by dustinswales** [comment]
@grantfirl Hmmm I don't see any unexpected changes [here](https://github.com/NCAR/ccpp-physics/pull/1069/files). I started w/ main, not on top of #439.

**Comment by grantfirl** [comment]
> @grantfirl Hmmm I don't see any unexpected changes [here](https://github.com/NCAR/ccpp-physics/pull/1069/files). I started w/ main, not on top of #439.

@dustinswales Ya, I think it was the fact that ccpp-physics had merges that didn't need SCM updates. I've merged in #439, so you would please pull in main and commit/push one more time, please (also for https://github.com/NCAR/ccpp-physics/pull/1069)?

File: scm/etc/scripts/UFS_IC_generator.py [hunk 1]
```diff
@@ -2704,8 +2712,11 @@ def main():
     
     # get UFS IC data (TODO: flag to read in RESTART data rather than IC data and implement
     # different file reads)
-    (state_data, surface_data, oro_data) = get_UFS_IC_data(in_dir, grid_dir, tile, tile_i,\
-                                                           tile_j, old_chgres, lam)
+    (state_data, surface_data, oro_data, error_msg) = get_UFS_IC_data(in_dir, grid_dir, tile, tile_i,\
+                                                                      tile_j, old_chgres, lam)
+    if (error_msg):
+        print("ERROR: unknown grid orintation")
```
Comment (grantfirl) [inline scm/etc/scripts/UFS_IC_generator.py:2718]: spelling

Comment (grantfirl) [reply scm/etc/scripts/UFS_IC_generator.py:2718]: Plus, shouldn't we just print `error_msg`?

**Review by grantfirl — APPROVED** [review]

**Comment by grantfirl** [comment]
@dustinswales This looks fine to me. Only comment that may need to be addressed is the error_msg one. The other one might benefit from an additional inline code comment for clarification.

File: scm/etc/scripts/UFS_IC_generator.py [hunk 2]
```diff
@@ -1171,6 +1202,12 @@ def get_UFS_surface_data(dir, tile, i, j, old_chgres, lam):
     # fractional grid
     tiice_in = read_NetCDF_surface_var(nc_file, 'tiice', i, j, old_chgres, missing_variable_ice_layers)
 
+    # soil color
+    if (slmsk_in == 1):
+        scolor_in = 4
```
[the remark this answers is not in the document]
Comment (dustinswales) [reply scm/etc/scripts/UFS_IC_generator.py:1207]: This comes from the [FV3 IO routine, sfc_io_apply_safeguards()](https://github.com/NOAA-EMC/fv3atm/blob/develop/io/fv3atm_sfc_io.F90#L1509).
I will add this to the code.

**Comment by grantfirl** [comment]
@dustinswales I think you need to merge ccpp-physics main into https://github.com/NCAR/ccpp-physics/pull/1069, then point to it here to fix the CI failures.

**Comment by dustinswales** [comment]
> @dustinswales I think you need to merge ccpp-physics main into [NCAR/ccpp-physics#1069](https://github.com/NCAR/ccpp-physics/pull/1069), then point to it here to fix the CI failures.

Ugh, I forgot to push my updated physics before updating the ccpp-scm. CI cranking along now. Sorry.

**Comment by grantfirl** [comment]
@dustinswales OK, time to revert .gitmodules and point to the latest ccpp-physics main: https://github.com/NCAR/ccpp-physics/commit/6d8fccbea12c11387c0bc457dcb3855574cd29aa

**Comment by grantfirl** [comment]
Thanks @dustinswales

**Pull request merged**